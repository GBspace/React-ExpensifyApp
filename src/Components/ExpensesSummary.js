import React from 'react';
import {connect} from 'react-redux';
import selectExpenses from '../selectors/expenses';
import expensesTotal from '../selectors/expenses-total';
import {Link} from 'react-router-dom';

export class ExpenseSummary extends React.Component{
    render(){
        const expenseLabel = this.props.expenseCount === 1 ? 'expense' : 'expenses';
        return(
            <div className="page-header">
                <div className = "content-container">
                    <h1 className="page-header__title">Viewing <span>{this.props.expenseCount} </span> {expenseLabel} totalling to <span>{this.props.expenseTotal}</span> </h1>
                    <div className="page-header__actions">
                        <Link className = "button" to="/create">Add Expense</Link> 
                    </div>
                </div>
            </div>
        );
    };
};



const mapStateToProps = (state)=>{
    let visibleExpenses = selectExpenses(state.expense,state.filters);
    return{
        expenseCount: visibleExpenses.length,
        expenseTotal: expensesTotal(visibleExpenses)
    };
};

export default connect(mapStateToProps)(ExpenseSummary);