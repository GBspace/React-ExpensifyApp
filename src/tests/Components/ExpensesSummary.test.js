import React from 'react';
import {shallow} from 'enzyme';
import ExpensesSummary from '../../Components/ExpensesSummary';

test('should render expenses summary with one expense',()=>{
    const wrapper = shallow(<ExpensesSummary expenseCount={1} expenseTotal={20}/>);
    expect(wrapper).toMatchSnapshot();

});

test('should render expenses summary with multiple expense',()=>{
    const wrapper = shallow(<ExpensesSummary expenseCount={23} expenseTotal={2000}/>);
    expect(wrapper).toMatchSnapshot();
});