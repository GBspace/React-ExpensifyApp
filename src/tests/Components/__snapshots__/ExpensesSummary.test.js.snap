// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render expenses summary with multiple expense 1`] = `
<div
  className="page-header"
>
  <div
    className="content-container"
  >
    <h1
      className="page-header__title"
    >
      Viewing 
      <span>
        23
         
      </span>
       
      expenses
       totalling to 
      <span>
        2000
      </span>
       
    </h1>
    <div
      className="page-header__actions"
    >
      <Link
        className="button"
        replace={false}
        to="/create"
      >
        Add Expense
      </Link>
    </div>
  </div>
</div>
`;

exports[`should render expenses summary with one expense 1`] = `
<div
  className="page-header"
>
  <div
    className="content-container"
  >
    <h1
      className="page-header__title"
    >
      Viewing 
      <span>
        1
         
      </span>
       
      expense
       totalling to 
      <span>
        20
      </span>
       
    </h1>
    <div
      className="page-header__actions"
    >
      <Link
        className="button"
        replace={false}
        to="/create"
      >
        Add Expense
      </Link>
    </div>
  </div>
</div>
`;
